includes/reports/overview/data-provider/class-sensei-reports-overview-data-provider-students.php

```diff
@@ -23,8 +44,90 @@ class Sensei_Reports_Overview_Data_Provider_Students implements Sensei_Reports_O
 	 * @return array
 	 */
 	public function get_items( array $filters ): array {
-		// TODO: Implement get_items() method.
-		return [];
+		$this->date_from = $filters['last_activity_date_from'] ?? null;
+		$this->date_to   = $filters['last_activity_date_to'] ?? null;
+
+		$query_args = array(
+			'fields' => [ 'ID', 'user_login', 'user_email', 'user_registered', 'display_name' ],
+		);
+
+		if ( ! empty( $filters['search'] ) ) {
+			$query_args['search'] = '*' . trim( $filters['search'], '*' ) . '*';
+		}
+
+		/**
+		 * Filter the WP_User_Query arguments
+		 *
+		 * @since 1.6.0
+		 * @param $query_args
+		 */
+		$query_args = apply_filters( 'sensei_analysis_overview_filter_users', $query_args );
+
+		add_action( 'pre_user_query', [ $this, 'add_last_activity_to_user_query' ] );
+		add_action( 'pre_user_query', [ $this, 'filter_users_by_last_activity' ] );
+		$wp_user_search = new WP_User_Query( $query_args );
```

Comment:
Are we missing the offset, per-page count and other query parameters here?

Reply:
Yep, nice catch!